test/suite/test_verify_disagg.py

```diff
@@ -0,0 +1,132 @@
+#    (we already have an OpLog imitation in some tests for layered tables)
+
+@disagg_test_class
+class test_verify_disagg(wttest.WiredTigerTestCase, DisaggConfigMixin):
+    hs = [
+        ('empty', dict(fill_hs=False)),
+        ('populated', dict(fill_hs=True)),
+    ]
+    disagg_storages = gen_disagg_storages('test_verify_disagg', disagg_only = True)
+    scenarios = make_scenarios(hs, disagg_storages)
+
+    nitems = 10000
+    timestamp = 2
+
+    conn_base_config = 'disaggregated=(page_log=palm),'
+    conn_config = conn_base_config + 'disaggregated=(role="leader")'
+    conn_config_follower = conn_base_config + 'disaggregated=(role="follower")'
+
+    table_cfg = 'key_format=S,value_format=S,block_manager=disagg'
+
+    session_follow = None
+    conn_follow = None
+
+    uri = 'layered:test_verify_disagg'
+
+    def leader_put_data(self, ts, value_prefix = '', low = 1, high = nitems):
+        cursor = self.session.open_cursor(self.uri, None, None)
+        for i in range(low, high):
+            self.session.begin_transaction()
+            cursor[str(i)] = value_prefix + str(i)
+            self.timestamp += 1
+            ts_cfg = "commit_timestamp=" + self.timestamp_str(self.timestamp) if ts else None
+            self.session.commit_transaction(ts_cfg)
+        cursor.close()
+
+    def verify(self, sessions, err_msg = None):
+        for session in sessions:
+            try:
+                session.verify(self.uri)
+            except wiredtiger.WiredTigerError as e:
+                if (err_msg):
+                    self.assertTrue(err_msg == str(e))
+                else:
+                    raise(e)
+
+    def create_follower(self):
+        self.conn_follow = self.wiredtiger_open('follower', self.extensionsConfig() + ',create,' +
+                                                self.conn_config_follower)
+        self.session_follow = self.conn_follow.open_session('')
+
+    def test_verify_disagg(self):
+        if self.fill_hs:
+            self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(self.timestamp))
+
+        # Create a table in the leader
+        self.session.create(self.uri, self.table_cfg)
+        # Verify the empty leader's table
+        self.verify([self.session])
+
+        # Create a follower
+        self.create_follower()
+        # The leader's table stays empty, the follower creation doesn't mean loading tables from the leader (it requires reconfiguration)
+        self.verify([self.session])
+        self.verify([self.session_follow], "No such file or directory")
+
+        # Create an empty checkpoint
+        self.session.checkpoint()
+        self.verify([self.session])
+
+        self.disagg_advance_checkpoint(self.conn_follow)
+        # Now both connections should have empty tables
+        self.verify([self.session, self.session_follow])
+
+        # Put some data to the leader
+        self.leader_put_data(self.fill_hs)
```

Comment:
The first arg to leader_put_data is a timestamp...?  Will a later ticket fix the test?

Reply:
I made this intentionally and the idea there was that "if we need to fill the HS, we need to specify a timestamp, which causes WT to insert entries to the HS" so it did sound meaningfull to me. However, I noticed that we can use `fill_hs` directly in the function, so I removed the argument. Please check the last commit.